src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/EnumerableExtensions.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions
+{
+    internal static class EnumerableExtensions
+    {
+        public static IEnumerable<T> Shuffle<T>(this IEnumerable<T> values)
+        {
+            var rdm = new Random();
+            T[] elements = values.ToArray();

```

Comment:
That's not proper usage of an enumerable. We shouldn't need to build an array.

Reply:
But seems we can't have a genuine random algorithm before we going through the entire enumerable